types.go

```diff
@@ -21,17 +23,17 @@ type Transaction struct {
 } // TODO Handle action details
 
 type Block struct {
-	Author           string        `json:"author"`
-	ExtraData        []int         `json:"extraData"`
-	Hash             string        `json:"hash"`
-	Number           int           `json:"number"`
-	ParentHash       string        `json:"parentHash"`
-	Score            string        `json:"score"`
-	Seal             [][]int       `json:"seal"`
-	StateRoot        string        `json:"stateRoot"`
-	Timestamp        int           `json:"timestamp"`
-	Transactions     []Transaction `json:"transactions"`
-	TransactionsRoot string        `json:"transactionsRoot"`
+	Author           json.RawMessage `json:"author"`
```

Comment:
Using string value in the `Block` type seems good. This `Block` data will be used as the return value of RPCs. The Author, Hash, etc data will be delivered in the string type encoded as hex.

@gurrpi @cubismic What do you think about it?

Reply:
I think using `string` in some case is better than `json.RawMessage`.
Because `json.Rawmessage` should be deserialized at some point and I think user should access to the elements of `Block` easily.

Reply:
And how about using `Hash` fields as `[]byte` type?
Every hash function defined in `crypto` package in this PR is using `[]byte` types.

Reply:
> I think using `string` in some case is better than `json.RawMessage`.
> Because `json.Rawmessage` should be deserialized at some point and I think user should access to the elements of `Block` easily.

I also think `string` type is better for usability perspective.

Reply:
> @gurrpi , @majecty
> And how about using `Hash` fields as `[]byte` type?
> Every hash function defined in `crypto` package in this PR is using `[]byte` types.

IMO, Hash functions serves for computation and its input/output is simple binary data; which is not mainly related to hexstring.

Reply:
@cubismic @gurrpi 
I don't know which one is better between using `string` or `[]byte`. If we use `[]byte` type, we have to implement error handling(when the data is not decoded by hex string format) and implement some complex serialize/deserialize functions. What do you think about delay using the `[]byte` type until it is needed?

Reply:
I think delaying is good option. btw, we should discuss which one to use `string` or `[]byte` more clearly. I'm also confused and can't make my side now.

Reply:
@majecty Well, I think error handling is required for current implemented hash functions. Because it uses `hex.hexDecodeString(s string)` and this decode function assumes _`string` contains only hexadecimal characters and has even length_ ([docs](https://golang.org/pkg/encoding/hex/#DecodeString)).

When we use `[]byte` than the possible use case of the function `Blake256(data []byte) []byte` would be like below: 
```
str := "abcd"
var rawData []byte = hex.DecodeString(str)
var hash string = hex.EncodeToString(Blake256(rawData))
```
So its just enought for `[]byte` and there is no need for additional implementation task. 🥳Please let me know if i have misunderstanding.

Reply:
@gurrpi 

Let's split subjects.

1. Using `string` or `[]byte` type in the `Block` type's `Hash` field.

I want to use `string` type here. We are using `JSON-RPC` protocol. We receive JSON data from a CodeChain node. If we use the `string` type, we can return the data we received from the CodeChain directly. However, if we use `[]byte` type, it is not that easy. Original `Hash` value is binary data. In the JSON response, the `Hash` value is encoded as hex-encoding. To use the `[]byte` type we should decode the string using hex-encoding, and the decoding function could fail. IMO, we are implementing the alpha version of the SDK, let's make the SDK simple, small, and working and improve the SDK later.

2. Error handling in Hash functions(Blake256, ...)

I agree with you. We should return errors instead of ignoring them.

Since the conversation is long, We may have a hangout call to make a decision. What do you think about it? @gurrpi @cubismic

Reply:
How about doing hangout call at 14:30 KST tomorrow? I could afford time usually after 14:00 KST in weekday.

Reply:
I'm OK. @cubismic said that he will ok at the time. @gurrpi Could you give me an email? I'll send a notification to you.

Reply:
@majecty Good. Here is my email: paik@a30a.dev

Reply:
@gurrpi I sent an invitation to you. Please check it.